cardano-api/src/Cardano/Api/Types.hs

```diff
@@ -104,6 +108,11 @@ data Address
   deriving (Eq, Generic , NFData, Show)  -- Byron.Address' needs NFData
   deriving NoUnexpectedThunks via UseIsNormalForm Address
 
+
+data Certificate
+  = ByronCertificate
```

Comment:
I think we can drop the byron case, there are no such things (at least not in a tx), and make this a newtype.